### ISSUE TITLE
YYC mishandles struct literals keys with spaces

### ISSUE THREAD

**Issue by Alphish** [YoYoGames/GameMaker-Bugs#2134]
### Description
If you define a struct literal with string key notation, and the key has a space, the space is replaced with underscore on YYC (Windows YYC at least).
Struct literal keys with spaces work correctly on Windows VM.

### Steps To Reproduce
Given the following piece of code:
```gml
var _test = {
    "lorem": 123,
    "ipsum dolor": 456,
};
show_debug_message(struct_get_names(_test));
show_debug_message(_test[$ "lorem"]);
show_debug_message(_test[$ "ipsum dolor"]);
show_debug_message(_test[$ "ipsum_dolor"]);
```

On VM it will be correctly printes as:
```
[ "lorem","ipsum dolor" ]
123
456
undefined
```

On YYC it will instead be printed as:
```
[ "lorem","ipsum_dolor" ]
123
undefined
456
```

[595aea80-393d-4efd-9035-ada1d80893bf](https://api.gamemaker.io/api/github/downloads/595aea80-393d-4efd-9035-ada1d80893bf)

**Comment by rwkay** [comment]
Fixed in 2023.11

**Comment by mgeddesGM** [comment]
verified as of  IDE v2023.1100.0.437  Runtime v2023.1100.0.450
